### ISSUE TITLE
[FEA] Improved error reporting on libcudf code

### ISSUE THREAD

**Issue by mrocklin** [rapidsai/cudf#830]
```python
import numpy as np
import cudf
df = cudf.DataFrame({'x': np.ones(1000000)})
df.merge(df)
```

```python-traceback
---------------------------------------------------------------------------
GDFError                                  Traceback (most recent call last)
<ipython-input-4-d82f292bdbeb> in <module>()
----> 1 df.merge(df)

~/cudf/python/cudf/dataframe/dataframe.py in merge(self, other, on, how, lsuffix, rsuffix, type, method)
   1106         # Compute merge
   1107         cols, valids = cpp_join.join(lhs._cols, rhs._cols, on, how,
-> 1108                                      method=method)
   1109
   1110         # Output conversion - take cols and valids from `cpp_join` and

~/cudf/python/cudf/bindings/join.pyx in cudf.bindings.join.join()

~/cudf/python/cudf/bindings/join.pyx in cudf.bindings.join.join()

~/cudf/python/cudf/bindings/cudf_cpp.pyx in cudf.bindings.cudf_cpp.check_gdf_error()

GDFError: b'GDF_MEMORYMANAGER_ERROR'
```

It would be nice to have a `MemoryError` or something similar.  The error message might also be nicer.  It could be a string rather than bytes, and not all caps.

**Comment by kkraus14** [comment]
This would be dependent on #862

**Comment by kkraus14** [comment]
This is resolved by #862
